plugins/modules/dynamodb_table.py

```diff
@@ -870,9 +855,6 @@ def _update_table(current_table):
                     additional_global_index_changes=additional_global_index_changes,
                 )
 
-    if module.params.get("wait"):
```

Comment:
Not needed, the wait logic is going to be called again in the only place that `_update_table()` is called.